searx/search/processors/online.py

```diff
@@ -248,6 +248,10 @@ def get_default_tests(self):
                 'matrix': {'query': 'paris', 'lang': 'en'},
                 'result_container': ['not_empty', ('has_language', 'en')],
             }
+            tests['lang_de'] = {
```

Comment:
This describe a new test for all the engines that support the `language` parameter (I haven't run the code, just read the code for now).

For information, it is possible to add some custom tests to one engine. Just declare the global `additional_tests` variable inside the engine code:

```python
additional_tests = {
  'lang_de': {
      'matrix': {'query': 'paris', 'lang': 'de'},
      'result_container': ['not_empty', ('has_language', 'de')],
  }
}
```

Reply:
> For information, it is possible to add some custom tests to one engine. 

Ah, thanks for the hint .. I have to learn more about checker .. 

> This describe a new test for all the engines that support the language parameter (I haven't run the code, just read the code for now).

Is it a problem to have a test with lang 'de' on all engines? .. I know from articles, that there are many searx user from de.

Reply:
>  I have to learn more about checker ..

For now the code is the documentation, far from ideal for this feature.

>Is it a problem to have a test with lang 'de' on all engines?

The `lang_de` test fails with some other engines.
For example `!qwant :de time` returns mostly English results, `!qwant :de Deutsch` seems to be better.

I think it requires some back and forth between the engine code and the checker tests. 

For now, I feel it is better (?) to move the `lang_de` test to google.py ? (and import the `additional_tests` variable in the other google engines). Later we can generalize / adjust the test for the other engines.

Reply:
> For now the code is the documentation, 

It is OK for now .. I'm currently dealing with it .. if I have more experience, I can possibly document something 

> I think it requires some back and forth between the engine code and the checker tests.

That's right .. tests needs to be balanced between development and maintenance. E.g the google engine is bot sensitive and releases very fast CAPTCHA issues.  I added some additional test but commented them out for now, see 7121d995. They have their value when it comes to deep tests in the development cycle.